app/src/main/java/com/google/android/samples/socialite/ui/camera/CameraViewModel.kt

```diff
@@ -93,12 +98,42 @@ class CameraViewModel @Inject constructor(
         .setQualitySelector(QualitySelector.from(Quality.HIGHEST))
         .build()
 
-    private val videoCaptureUseCase = VideoCapture.Builder(recorder)
-        .build()
-
     private var currentRecording: Recording? = null
     private lateinit var recordingState: VideoRecordEvent
 
+    init {
+        val videoCaptureBuilder = VideoCapture.Builder(recorder)
+
+        viewModelScope.launch {
+            val hdrCameraInfo = getHdrCameraInfo()
+
+            if (hdrCameraInfo != null) {
+                Log.i(TAG, "Capturing HDR video")
+                videoCaptureBuilder.setDynamicRange(hdrCameraInfo)
+            }
+
+            videoCaptureUseCase = videoCaptureBuilder.build()
+        }
+    }
+
+    private suspend fun getHdrCameraInfo(): DynamicRange? {
+        var supportedHdrEncoding: DynamicRange? = null
+
+        cameraProviderManager.getCameraProvider().availableCameraInfos
+            .first { cameraInfo ->
+                val videoCapabilities = Recorder.getVideoCapabilities(cameraInfo)
+                val supportedDynamicRanges =
+                    videoCapabilities.supportedDynamicRanges
+
+                supportedHdrEncoding = supportedDynamicRanges.firstOrNull {
+                    it != DynamicRange.SDR // Ensure an HDR encoding is found
```

Comment:
Can we actually make this more specific as well?

There seem to be multiple different dynamic range profiles and to make it consistent, we should choose one we know they all support, which would be HLG 10-bit

https://developer.android.com/reference/androidx/camera/core/DynamicRange

Any device that supports HDR should support HLG 10-bit at a minimum

Reply:
done!